### PR TITLE
feat: stacking pool members endpoint

### PR DESCRIPTION
**Pull request by zone117x**
Closes https://github.com/hirosystems/stacks-blockchain-api/issues/465

TODO:
* [x] Unambiguous terminology for the endpoint, e.g. `GET /extended/beta/stacking/{pool_principal}/delegations`
* [x] Ability to query for recent/latest delegations, e.g. with a query param `?after_block=<block_height>`
* [x] Exclude old delegation events for the same stacker, e.g. with max block height per stacker

---

Implements a new endpoint that returns a list of stacking pool members for a given delegator principal. 

The endpoint is `GET /extended/beta/stacking/{pool_principal}/delegations` (notice the `beta` version prefix). This means the endpoint can have breaking changes in the future. Once we are comfortable with the shape, the endpoint will graduate to the `/extended/v1/` prefix.

Docs: https://deploy-preview-1592--stacks-blockchain-api.netlify.app/#tag/Stacking

Example usage:
```jsonc
// GET /extended/beta/stacking/SP3KWQTR2K5HD1A99BZ9XGAMFYBVT3QQ0NH7XTDP7/delegations?limit=200&offset=0

{
  "limit": 200,
  "offset": 0,
  "total": 2,
  "results": [
    {
      "stacker": "SPDGWQ60E1QXTC6E4VHKK0NG34EAX554WDFFFHMF",
      "pox_addr": "bc1qxhmdufsvnuaaaer4ynz88fspdsxq2h9e9cetdj",
      "amount_ustx": "2062697998",
      "burn_block_unlock_height": 781984,
      "block_height": 65917,
      "tx_id": "0x3c81cd26ed172a7e552ed245b8b290abefa4f82dfc610a0091ac91c2eed7f9e6"
    },
    {
      "stacker": "SP7RNXF7YBSS43TECXTHF7W1MAB221WWW8RKPGSN",
      "pox_addr": "1Di1YoMov6Ua3gPedfQz7TkP6iTLqbPUzi",
      "amount_ustx": "4815793194",
      "block_height": 65910,
      "tx_id": "0x1692cd26ed172a7e552ed245b8b290abefa4f82dfc610a0091ac91c2eed7f9e6"
    }
  ]
}
```

### CODE REVIEW

**Comment by friedger** [comment]
This looks good to me!

File: src/datastore/pg-store.ts [hunk 1]
```diff
@@ -2029,6 +2030,49 @@ export class PgStore {
...
+        {
+          stacker: string;
+          pox_addr: string | null;
+          amount_ustx: string;
+          burnchain_unlock_height: number | null;
+          tx_id: string;
+          total_rows: number;
+        }[]
+      >`
+        SELECT
```
[the remark this answers is not in the document]
Comment (zone117x) [reply src/datastore/pg-store.ts:2051]: Good call, looking into making that change

[the remark this answers is not in the document]
Comment (zone117x) [reply src/datastore/pg-store.ts]: @friedger this has been implemented via the query param `?after_block=<block_height>`

**Comment by friedger** [comment]
:shipit:

**Pull request merged**